ONSdigital/response-operations-ui
# No reply to deleted user

# Motivation and Context
Once users are deleted, we need to make it so that it's not possible to add to the conversation or modify its open/closed status

# What has changed
When a respondent cannot be found, the ability to modify the conversation is replaced with a message informing the user this cannot be done. 
Additional logic has been added to handle the possibility of someone hitting the endpoint with the correct information to bypass this restriction with appropriate error messages appearing.

# How to test?
*To test the removal of the reply functionality*
- Run acceptance tests to seed data
- Add conversation between respondent and internal user. Add 1 or 2 messages each side
- Delete the user `curl -v -d '{"email": "<your-email-here>"}' -H "Content-Type: application/json" -X DELETE http://admin:secret@localhost:8081/party-api/v1/respondents/email`
- Go to the conversation and see the reply box removed.

*To test the additional logic*
- Open templates/conversation-view/conversation-view.html
- Remove the if/else for checking for the deleted user variable
- Go to the conversation, the reply box should be there again
- Attempt to send a message.  You should see an error saying it can't be sent because the user has been deleted

# Links
https://trello.com/c/3TkwE4E8


# Screenshots (if appropriate):
